server/src/main/java/org/apache/kafka/server/config/AbstractKafkaConfig.java

```diff
@@ -79,7 +80,13 @@ public AbstractKafkaConfig(ConfigDef definition, Map<?, ?> originals, Map<String
     }
 
     public List<String> logDirs() {
-        return Csv.parseCsvList(Optional.ofNullable(getString(ServerLogConfigs.LOG_DIRS_CONFIG)).orElse(getString(ServerLogConfigs.LOG_DIR_CONFIG)));
+        return Optional.ofNullable(getList(ServerLogConfigs.LOG_DIRS_CONFIG))
+                .orElse(Stream.of(getString(ServerLogConfigs.LOG_DIR_CONFIG))
+                        .map(String::trim).filter(s -> !s.isEmpty()).toList()
```

Comment:
It's probably better to keep the empty string. Otherwise, we could fail with IllegalArgumentException when accessing an empty list, which is less intuitive.

Also, if we want to do `trim()`, it's probably better to do this consistently on all configs.

Reply:
I think we still need to split the LOG_DIRS_CONFIG, even if it only contains a single value. The previous logic parsed the LOG_DIR_CONFIG string and split it using \s*,\s*, and there is a test case that verifies this scenario.

https://github.com/apache/kafka/blob/0202721b4c175c606c3d079e602fe33e8faea018/core/src/test/scala/unit/kafka/server/KafkaConfigTest.scala#L1553